packages/wallet/api/src/lib-wallet.js

```diff
@@ -1790,6 +1791,29 @@ export function makeWalletRoot({
     serialize: context.serialize,
   });
 
+  const handleAcceptOfferAction = async offer => {
+    const { requestContext } = offer;
+    const rawId = await addOffer(offer, requestContext);
+    const dappOrigin =
+      requestContext.dappOrigin || requestContext.origin || 'unknown';
+    return acceptOffer(`${dappOrigin}#${rawId}`);
+  };
+
+  const handleSuggestIssuerAction = ({ petname, boardId }) =>
+    suggestIssuer(petname, boardId);
+
+  const performAction = async obj => {
+    const { type, data } = JSON.parse(obj.action);
```

Comment:
Coordinating around what goes in `obj.action` could be tricky, but using a `type` descriminator leaves plenty of room for evolution, so I suppose this is fine.

@michaelfig I wonder if you have any thoughts.